xarray/core/dataset.py

```diff
@@ -1611,6 +1611,11 @@ def __getitem__(
                 return self._construct_dataarray(key)
             except KeyError as e:
                 message = f"No variable named {key!r}. Variables on the dataset include {shorten_list_repr(list(self.variables.keys()), max_items=10)}"
+
+                best_guess = utils.did_you_mean(key, self.variables.keys())
```

Comment:
Amazing idea. I would print the best guess first, and then any others so that's it's easy to see

Reply:
Maybe should just remove the "Variables on the dataset include ..." ? They try to do the same thing I think.

Reply:
Yeah you could sort the whole list by similarity and then print that (truncated as above)

Reply:
Now it prioritizes best_guess. If best_guess is empty you could be working in the wrong dataset, so it's still nice to get some kind of clue which dataset you're using.